plugins/ui/src/js/src/WidgetTypes.ts

```diff
@@ -0,0 +1,29 @@
+import { WidgetDefinition } from '@deephaven/dashboard';
+import { ExportedObject } from './ElementUtils';
+
+export interface WidgetMessageDetails {
+  getDataAsBase64(): string;
+  getDataAsString(): string;
+  exportedObjects: ExportedObject[];
+}
+
+export interface WidgetMessageEvent {
+  detail: WidgetMessageDetails;
+}
+
+export interface JsWidget extends WidgetMessageDetails {
+  addEventListener: (
+    type: string,
+    listener: (event: WidgetMessageEvent) => void
+  ) => () => void;
+  sendMessage: (message: string, args: unknown[]) => void;
```

Comment:
I think I suggested this in another PR so we don't accidentally send references without knowing the actual type
```suggestion
  sendMessage: (message: string, references?: never[]) => void;
```